frontendGraph/src/main/java/com/virtuslab/gitmachete/frontend/graph/nodes/BaseGraphNode.java

```diff
@@ -59,4 +55,17 @@ public void setNextSiblingNodeIndex(@Positive int i) {
     assert nextSiblingNodeIndex == null : "nextSiblingNodeIndex has already been set";
     nextSiblingNodeIndex = i;
   }
+
+  @Override
+  public final boolean equals(@Nullable Object other) {
+    return this == other;
+  }
+
+  @Override
+  public final int hashCode() {
```

Comment:
Nit: ofc you can leave it now here... but actually given that we compare the instances using the same criterion as `Object#equals`, you could as well just return `super.hashCode()` here